scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt

```diff
@@ -87,6 +87,16 @@ internal class ScenarioRunServiceImpl(
       throw CsmAccessForbiddenException("You are not allowed to delete this Resource")
     }
     cosmosTemplate.deleteEntity("${organizationId}_scenario_data", scenarioRun)
+
+    deleteDataFromScenarioRunId(organizationId, scenarioRun.workspaceKey!!, scenariorunId)
```

Comment:
Please consider handling the case where data deletion does not succeed in ADX.

Reply:
C'est deja le cas dans AzureDataExplorerClient.kt non?

Reply:
Nope. I mean, if the call to `deleteDataFromScenarioRunId` errors, there is currently no way to rollback. This would return an error to the client, and leave the system in a state where the ScenarioRun no longer exists but there are still data related to it in ADX. Worse, clients would think the ScenarioRun was not deleted, while the document actually was.

I suggest you implement such rollback mechanism in a simple way, for now simply by catching potential exceptions that might be thrown by `deleteDataFromScenarioRunId` and restoring the ScenarioRun in such exceptional cases.

Reply:
Also, I am curious about the timing of the call to ADX. Is data deletion synchronous or not ? If yes, this might have an impact on the API response time.